src/js/calendar/CalendarController.js

```diff
@@ -0,0 +1,144 @@
+import TaskController from "../board/Task/TaskController";
+import {
+  $previousMonthButton,
+  $nextMonthButton,
+  $todayDayText,
+  $todayDateText,
+  $todayMonthText,
+  $todayYearText,
+  $selectedDayText,
+  $selectedDateText,
+  $selectedMonthText,
+  $selectedYearText,
+  $allDatesInCalendar,
+  $header,
+} from "../elements";
+import {
+  handleMoveMonthButton,
+  createHandleClickDateOfCalendar,
+} from "./calendarElementEventHandler";
+import { DAYS } from "./day";
+import Month from "./Month";
+import Now from "./Now";
+import SelectedDate from "./SelectedDate";
+
+export default class CalendarController {
+  #calendar;
+  #dateOfCalendarClickEventListenerRepository = {};
+
+  constructor(calendar) {
+    this.#calendar = calendar;
+    TaskController.setCalendarController(this);
+  }
+  initialize() {
+    this.#addEventListenerToMoveButton();
+    this.#printHeader();
+    this.displayCalendarContents();
+    this.displaySelectedDateText();
+  }
+
+  #addEventListenerToMoveButton() {
+    $previousMonthButton.addEventListener("click", handleMoveMonthButton(this));
+    $nextMonthButton.addEventListener("click", handleMoveMonthButton(this));
+  }
+
+  displayCalendarContents() {
+    if (
+      !this.#calendar.checkYearExistenceByDateObject(
+        SelectedDate.getDateObject()
+      )
+    ) {
```

Comment:
조건문이 긴 경우, 별도의 변수로 선언하여 작성하면 가독성을 높일 수 있습니다.
```suggestion
    const isYearExisted = !this.#calendar.checkYearExistenceByDateObject(SelectedDate.getDateObject());
    if (isYearExisted)
```